src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs

```diff
@@ -0,0 +1,194 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DelayedDelivery;
+    using DeliveryConstraints;
+    using Faults;
+    using Hosting;
+    using Logging;
+    using ObjectBuilder;
+    using Settings;
+    using Support;
+    using Transport;
+
+    class RecoverabilityComponent
+    {
+        public RecoverabilityComponent(SettingsHolder settings)
+        {
+            this.settings = settings;
+
+            settings.SetDefault(NumberOfDelayedRetries, DefaultNumberOfRetries);
+            settings.SetDefault(DelayedRetriesTimeIncrease, DefaultTimeIncrease);
+            settings.SetDefault(NumberOfImmediateRetries, 5);
+            settings.SetDefault(FaultHeaderCustomization, new Action<Dictionary<string, string>>(headers => { }));
+            settings.AddUnrecoverableException(typeof(MessageDeserializationException));
+        }
+
+        public RecoverabilityExecutorFactory GetRecoverabilityExecutorFactory()
```

Comment:
I wonder if this should just be a property of type Lazy<RecoverabilityExecutorFactory>` instead of a get method

Reply:
Agree, done